src/components/utils/tekton-results.ts

```diff
@@ -358,6 +361,50 @@ export const getTaskRuns = (
   cacheKey?: string,
 ) => getFilteredTaskRuns(namespace, '', options, nextPageToken, cacheKey);
 
+export const createTektonResultsSummaryUrl = async (
+  namespace: string,
+  options?: TektonResultsOptions,
+  nextPageToken?: string,
+): Promise<string> => {
+  const tektonResult: K8sResourceKind = await k8sGet({
+    model: TektonResultModel,
+    name: 'result',
+  });
+  const targetNamespace = tektonResult?.spec?.targetNamespace;
+  const route: K8sResourceKind = await k8sGet({
+    model: RouteModel,
+    name: 'tekton-results-api-service',
+    ns: targetNamespace,
+  });
+  const tektonResultUrl = route?.spec.host;
+  if (!tektonResultUrl) {
+    throw new Error('route.spec.host is undefined');
+  }
+  const URL = `https://${tektonResultUrl}/apis/results.tekton.dev/v1alpha2/parents/${namespace}/results/-/records/summary?${new URLSearchParams(
+    {
+      summary: `${options?.summary}`,
+      // default sort should always be by `create_time desc`
+      // order_by: 'create_time desc', not supported yet
+      page_size: `${Math.max(
+        MINIMUM_PAGE_SIZE,
+        Math.min(
+          MAXIMUM_PAGE_SIZE,
+          options?.limit >= 0 ? options.limit : options?.pageSize ?? 30,
+        ),
+      )}`,
+      ...(nextPageToken ? { page_token: nextPageToken } : {}),
+      filter: AND(
+        EQ('data_type', options.data_type.toString()),
+        options.filter,
+        selectorToFilter(options?.selector),
+        options?.filter,
+        options?.groupBy,
```

Comment:
groupBy will be not always defined so it will add undefined.

Reply:
where to add it then?

Reply:
changed, please review